### PR TITLE
PY3: safeguard __del__ in Annex and Git Repo to not breed useless exceptions

### PR DESCRIPTION
**Pull request by yarikoptic**
as e.g. reported in https://github.com/datalad/datalad-crawler/issues/47

### CODE REVIEW

File: datalad/support/annexrepo.py [hunk 1]
```diff
@@ -350,6 +350,15 @@ def set_default_backend(self, backend, persistent=True, commit=True):
             self.config.set('annex.backends', backend, where='local')
 
     def __del__(self):
+
+        def safe__del__debug(e):
+            """We might be too late in the game and either .debug or exc_str
+            are no longer bound"""
```
Comment (kyleam) [inline datalad/support/annexrepo.py]: Only an AttributeError is caught below.  If exc_str() was no longer bound, wouldn't that be a NameError?

Comment (yarikoptic) [reply datalad/support/annexrepo.py]: good question... may be?  let me add it as well just in case ;-) thanks!

Comment (yarikoptic) [reply datalad/support/annexrepo.py]: although I do not think local binds disappear first... at least I can't recall if I ran into such before.  But shouldn't hurt - pushed 59d7d48a553ed8e2851a37ae0af839e28330d528

Comment (kyleam) [reply datalad/support/annexrepo.py]: OK, thanks.  If exc_str is not a known problem, of course it'd be fine to just remove the false information from the docstring.  But not a big deal either way.

Comment (yarikoptic) [reply datalad/support/annexrepo.py]: ah, I see !  thanks for spotting that -- I forgot that I wrote it ;)

**Pull request merged**